src/main/java/org/klodnicki/service/BookService.java

```diff
@@ -39,4 +40,23 @@ public void update(BookInfo bookInfo) {
     public List<BookInfo> findBooksByTitleAndAuthor(String title, String author) {
         return bookRepository.findBooksByTitleAndAuthor(title, author);
     }
+
+    public List<String> showBooks() throws NotFoundInDatabaseException {
+        return prepareListOfAllBooks();
+    }
+
+    private List<String> prepareListOfAllBooks() throws NotFoundInDatabaseException {
```

Comment:
This method can be improved in several ways:

- Getting List<BookInfo> could be separate method. 
- Checking if results are empty could be done before by checking if allBooksInDatabase is empty.
- This method should rather only prepare books by changing them to Strings, right now it does two main functions.

Reply:
- [x] getting list is a separate method now
- [x] checking if results are empty is now before main body
- [x] method prepare books without searching in db